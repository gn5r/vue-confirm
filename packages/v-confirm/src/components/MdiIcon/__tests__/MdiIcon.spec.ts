import { createLocalVue, mount, MountOptions } from "@vue/test-utils";
import Icon from "../MdiIcon";
import { mdiClose, mdiPencil } from "@mdi/js";

describe("MdiIcon.ts", () => {
  const localVue = createLocalVue();

  type Instance = InstanceType<typeof Icon>;
  const mountFunc = (options?: MountOptions<Instance>) =>
    mount(Icon, { localVue, ...options });

  it("should render mdi icon", () => {
    const wrapper = mountFunc({ slots: { default: [mdiClose] } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render button when having click listener", () => {
    const func = jest.fn(() => "hello");
    const wrapper = mountFunc({
      slots: { default: [mdiPencil] },
      listeners: {
        click: func,
      },
    });
    const button = wrapper.find("button");
    expect(wrapper.html()).toMatchSnapshot();
    expect(button.exists()).toBe(true);
  });

  it("render a dark component", () => {
    const wrapper = mountFunc({ propsData: { dark: true } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
