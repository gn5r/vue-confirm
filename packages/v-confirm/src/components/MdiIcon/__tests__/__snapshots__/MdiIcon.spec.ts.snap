// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MdiIcon.ts should render button when having click listener 1`] = `
<button type="button" style="width: 24px; height: 36px; background-color: transparent; cursor: pointer; outline: none; padding: 0px; appearance: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" area-hidden="true" style="font-size: 24px; width: 24px; height: 24px; fill: currentColor; vertical-align: middle;">
    <path d="M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z"></path>
  </svg></button>
`;

exports[`MdiIcon.ts should render mdi icon 1`] = `<span area-hidden="true" style="width: 24px; height: 36px; background-color: transparent; cursor: pointer; outline: none; padding: 0px; appearance: none;"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" area-hidden="true" style="font-size: 24px; width: 24px; height: 24px; fill: currentColor; vertical-align: middle;"><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"></path></svg></span>`;
